packages/nimble-components/src/table/specs/table-column-specs/table-column-mapping.md

```diff
@@ -0,0 +1,359 @@
+
+Pros:
+
+-   It would not require updates to the API if we needed to support new types of mapped content (e.g. icon with text), or if the mapped content itself got new configuration options (e.g. a scaling factor for icons).
+-   Undefined element types caught at compile time.
+
+Cons:
+
+-   Verbose. Requires user to create `template` element and wrap text in `span`s for styling purposes.
+-   Requires difficult validation to ensure only supported elements are present in the `template`.
+-   Could allow users to provide inline styling.
+-   Blazor: cannot put Blazor components inside `template`--must use raw Nimble elements without type safety
+
+### API
+
+#### Icon column element:
+
+_Component Name_
+
+-   `nimble-table-column-icon`
+
+_Props/Attrs_
+
+-   `field-name`: string
+-   `data-type`: 'string' | 'number' | 'boolean' | null (defaults to 'string')
+-   `pixel-width`: number (set to the desired fixed column width, else will use a default fixed width)
+
+_Content_
+
+-   column title (text or icon)
+-   1 or more `nimble-mapping-icon` or `nimble-mapping-spinner` elements
+
+#### General mapping column element:
+
+_Component Name_
+
+-   `nimble-table-column-mapping`
+
+_Props/Attrs_
+
+-   `field-name`: string
+-   `data-type`: 'string' | 'number' | 'boolean' | null (defaults to 'string')
+-   `fractional-width`: number (defaults to 1)
+-   `min-pixel-width`: number (defaults to minimum supported by table)
+
+_Content_
+
+-   column title (text or icon)
+-   1 or more `nimble-mapping-text` elements
+
+#### Mapping element (icon):
+
+_Component Name_
+
+-   `nimble-mapping-icon`
+
+_Props/Attrs_
+
+-   `key`: string | number | boolean | null
+-   `icon`: string - name of the Nimble icon element
+-   `severity`: string - one of the supported enum values. Controls color of the icon.
+-   `label`: string - localized value used as the accessible name and `title` of the icon. Will also be displayed in the group header.
+-   `default`: boolean - presence causes this mapping to be used when no others match the value
+
+#### Mapping element (spinner):
+
+_Component Name_
+
+-   `nimble-mapping-spinner`
+
+_Props/Attrs_
+
+-   `key`: string | number | boolean | null
+-   `label`: string - localized value used as the accessible name and `title` of the spinner. Will also be displayed in the group header.
+-   `default`: boolean - presence causes this mapping to be used when no others match the value
+
+#### Mapping element (text):
+
+_Component Name_
+
+-   `nimble-mapping-text`
+
+_Props/Attrs_
+
+-   `key`: string | number | boolean | null
+-   `label`: string - display text
+-   `default`: boolean - presence causes this mapping to be used when no others match the value
+
+### Anatomy
+
+#### `nimble-table-column-mapping`
```

Comment:
A couple thoughts:
1. this will have the logic to display a tooltip when text overflows, right? It's not necessary to copy it again in the spec, but would be good to call it out
2. hopefully this can reuse some of the components described in #1224. A goal of that HLD is that anything that renders as text (text columns, numeric columns, date time columns, etc) will render identically by sharing base classes / cell elements / header elements. I think we'd want that to apply to mapping columns that display text too.

Reply:
1. I didn't put the templates for the mapping text element in this section, but I did call this out in the design section:
> Text in a grouping header or in the cell will be ellipsized and gain a tooltip when the full text is too long to display.

2. I'll try to share whatever I can.